ReactSkia/sdk/WindowDelegator.cpp

```diff
@@ -61,16 +61,34 @@ void  WindowDelegator::createNativeWindow() {
 }
 
 void WindowDelegator::closeWindow() {
-  RNS_LOG_TODO("Sync between rendering & Exit to be handled ");
   windowActive = false;
-  std::scoped_lock lock(renderCtrlMutex);
 
   if(ownsTaskrunner_) {
+    windowTaskRunner_->dispatch([&](){
+      closeNativeWindow();
+      sem_post(&semReadyToDraw_);
+    });
+
+    sem_wait(&semReadyToDraw_);
     windowTaskRunner_->stop();
```

Comment:
Can't we do stop task loop from the task running in taskloop.
If that is possible then , at the end of "closeNativeWindow", stop can be placed and "closeWindow" will wait using "join".
By this need of semaphore synchronization can be avoided.

Reply:
yes , updated.